CorsixTH/Lua/dialogs/staff_dialog.lua

```diff
@@ -277,6 +285,11 @@ function UIStaff:onMouseUp(button, x, y)
       local staff = ui.hospital.staff[i]
       if staff.profile.humanoid_class == self.staff.profile.humanoid_class then
         ui:addWindow(UIStaff(ui, staff))
+        if hit_namebox then
+          local sx, sy = ui.app.map:WorldToScreen(staff.tile_x, staff.tile_y)
+          local dx, dy = staff.th:getPosition()
+          ui:scrollMapTo(sx + dx, sy + dy)
+        end
```

Comment:
If you wanted to go one step further, you could pull this out into a local function inside `UIStaff:onMouseUp`; possibly just before the `if button = right` line

```lua
local function scrollToMapPosition(staff)
  if not hit_namebox then return end
  local sx, sy = ui.app.map:WorldToScreen(staff.tile_x, staff.tile_y)
  local dx, dy = staff.th:getPosition()
  ui:scrollMapTo(sx + dx, sy + dy)
end
```
(I'm pretty sure inserting the function at the point I said means it is aware of `local ui` and `local hit_namebox`)

Then:
```lua
scrollToMapPosition(staff)
```